backend/src/main/java/gov/cdc/usds/simplereport/service/TestOrderService.java

```diff
@@ -126,24 +135,112 @@ public TestOrderService(
       if (result != null) {
         p = cb.and(p, cb.equal(root.get(BaseTestInfo_.result), result));
       }
+      if (role != null) {
+        p = cb.and(p, cb.equal(root.get(BaseTestInfo_.patient).get(Person_.role), role));
+      }
+      if (symptomatic != null) {
+        String noSymptomsStr =
+            symptomatic == OptionalBoolean.YES
+                ? "false"
+                : symptomatic == OptionalBoolean.NO
+                    ? "true"
+                    : symptomatic == OptionalBoolean.WOULD_NOT_SPECIFY ? "false" : "false";
+        p =
+            cb.and(
+                p,
+                cb.equal(
+                    cb.function(
+                        "jsonb_extract_path_text",
```

Comment:
I believe filtering on a `jsonb` field will cause performance issue in production

Reply:
I would be interested in seeing what the plan is for the query that is generated. You should be able to grab the query from the console logs and then run it manually against your local DB with `explain`